doc/reference/autoinstall-reference.rst

```diff
@@ -3,184 +3,1013 @@
 
-stub
+* **type:** list of strings
+* **default:** []
+
+A list of config keys to still show in the UI. So for example:
+
+.. code-block:: yaml
+
+    version: 1
+    interactive-sections:
+     - network
+    identity:
+     username: ubuntu
+     password: $crypted_pass
+
+Would stop on the network screen and allow the user to change the defaults. If
+a value is provided for an interactive section it is used as the default.
+
+You can use the special section name of "\*" to indicate that the installer
+should ask all the usual questions -- in this case, the :file:`autoinstall.yaml`
+file is not really an "autoinstall" file at all, instead just a way to change
+the defaults in the UI.
+
+Not all config keys correspond to screens in the UI. This documentation
+indicates if a given section can be interactive or not.
+
+If there are any interactive sections at all, the :ref:`ai-reporting` key is
+ignored.
 
 .. _ai-early-commands:
 
 early-commands
-==============
+--------------
 
-stub
+* **type:** :ref:`command list<ai-command-lists>`
+* **default:** no commands
+* **can be interactive:** no
 
-.. _ai-error-commands:
+A list of shell commands to invoke as soon as the installer starts, in
+particular before probing for block and network devices. The autoinstall
+config is available at :file:`/autoinstall.yaml` (irrespective of how it was
+provided) and the file will be re-read after the ``early-commands`` have run to
+allow them to alter the config if necessary.
 
-error-commands
-==============
+.. _ai-locale:
 
-stub
+locale
+------
 
-.. _ai-identity:
+* **type:** string
+* **default:** ``en_US.UTF-8``
+* **can be interactive:** yes, always interactive if any section is
 
-identity
-========
+The locale to configure for the installed system.
 
-stub
+.. _ai-refresh-installer:
 
-.. _ai-interactive-sections:
+refresh-installer
+-----------------
 
-interactive-sections
-====================
+* **type:** mapping
+* **default:** see below
+* **can be interactive:** yes
 
-stub
+Controls whether the installer updates to a new version available in the given
+channel before continuing.
 
-.. _ai-kernel:
+The mapping contains keys:
 
-kernel
-======
+update
+~~~~~~
+
+* **type:** boolean
+* **default:** ``false``
+
+Whether to update or not.
 
-stub
+channel
+~~~~~~~
+
+* **type:** string
+* **default:** ``"stable/ubuntu-$REL"``
+
+The channel to check for updates.
 
 .. _ai-keyboard:
 
 keyboard
-========
+--------
 
-stub
+* **type:** mapping, see below
+* **default:** US English keyboard
+* **can be interactive:** yes
 
-.. _ai-late-commands:
+The layout of any attached keyboard. Often systems being automatically
+installed will not have a keyboard at all in which case the value used here
+does not matter.
 
-late-commands
-=============
+The mapping's keys correspond to settings in the :file:`/etc/default/keyboard`
+configuration file. See
+`its manual page <http://manpages.ubuntu.com/manpages/bionic/en/man5/keyboard.5.html>`_
```

Comment:
This link needs to be updated, not sure what to update it to. I recommend a version that is somehow agnostic to release if possible, to reduce maintenance burden going forward.

Reply:
I love the idea of a release agnotic version but not sure where that would be.  Let's bump the URL to https://manpages.ubuntu.com/manpages/focal/en/man5/keyboard.5.html for now.

Reply:
Somehow I recently learned that https://manpages.ubuntu.com/manpages/keyboard.5.html works as a release agnostic version. It would be nice to have a "latest LTS" link but I don't know how any of this works!